ensime_shared/launcher.py

```diff
@@ -163,6 +163,13 @@ def _start_process(self, classpath):
         null = open(os.devnull, "r")
         java = os.path.join(self.config['java-home'], 'bin', 'java')
 
+        if(os.name == "nt"):
+            java += ".exe" #the java executable has the .exe extension in windows
+            # windows uses ';' instead of ':' for the classpath sep since ':' is reserved (C:\)
+            # for example  "c:\ensime.jar:c:\scala.jar"  =>  "c:\ensime.jar;c:\scala.jar"
+            # the following ".jar:" hack works because only jar's happen to be on the ensime classpath
+            classpath = classpath.replace(".jar:",".jar;")
```

Comment:
@drxcc well, I had a better look at where `classpath` is formatted from a list and it is done in multiple places, one per each launcher strategy, like in [this line](https://github.com/drxcc/ensime-vim/blob/f6abb669269646b097fb4765b3bc55d3de11e768/ensime_shared/launcher.py#L234)

I think the right thing to do here is to carry `classpath` as a list until it really needs to be formatted as string, that is right here in `_start_process` where you are doing it, but manipulating it as string seems a bit dirty :), so after changing it to be a list you can , you can pick the proper separator.

What you think? do you want to do the small refactor?

Please note that there is a tests for that which should be update accordingly. You can run the tests with `make test`